include/range/v3/detail/optional.hpp

```diff
@@ -86,6 +86,49 @@ namespace ranges
             }
         };
 
+        template<typename T>
+        struct optional<T &>
+        {
+        private:
+            variant<meta::nil_, T &> data_;
```

Comment:
True: you'd have to disable one of them, and turn all the `move`s into `forward`s. The fifteen-line specialization is probably simpler.